tests/components/tts/test_init.py

```diff
@@ -1399,8 +1399,33 @@ def test_resolve_engine(hass: HomeAssistant, setup: str, engine_id: str) -> None
     ):
         assert tts.async_resolve_engine(hass, None) is None
 
-    with patch.dict(hass.data[tts.DATA_TTS_MANAGER].providers, {"cloud": object()}):
-        assert tts.async_resolve_engine(hass, None) == "cloud"
+
+@pytest.mark.parametrize(
+    ("setup", "engine_id"),
+    [
+        ("mock_setup", "test"),
+        ("mock_config_entry_setup", "tts.test"),
+    ],
+    indirect=["setup"],
+)
+def test_resolve_engine_with_cloud(
+    hass: HomeAssistant, setup: str, engine_id: str
+) -> None:
+    """Test resolving engine."""
+    with patch.dict(
+        hass.data["tts"]._entities,
+        {
```

Comment:
How can there be entities if we don't set up the config entry in the test setup phase? This test isn't realistic.

Reply:
The unrealistic tests have been replaced with tests which add:
- A cloud entity
- A non cloud entity
- A legacy engine